### ISSUE TITLE
mfa_required_since_version may not mean what it implies

### ISSUE THREAD

**Issue by sonalkr132** [rubygems/rubygems.org#2964]
Consider the scenario:
```
* gem has 2.0.1, 3.0.1 published
* latest_version points to 3.0.1
* user publishes 2.0.2 with rubygems_mfa_required set
```
rubygem page will start saying "required mfa since 2.0.2" which is not true (3.0.1 was published without mfa). We can't enforce that version numbers are published in order.

## Expected Behavior
Perhaps we should not mention version number in "required since". I have replaced it with just "required mfa true" if the current version has mfa enabled.

## Current Behavior
Misleading "required mfa since". The function also blows up if latest_version doesn't have next. `mfa_required_since_version` expects that there is a public_version after the most recent non_mfa_version `non_mfa_version.next.number`, which may not always be the case.

## Possible Solution
?

## Additional Context
https://github.com/rubygems/rubygems.org/pull/2242

**Comment by jenshenny** [comment]
Hmm, would the date of when the mfa requirement was pushed work instead of the version number? 

Eg if user publishes `2.0.2` with `rubygems_mfa_required` set on March 24, 2022. The heading would be `since March 24, 2022`. It's more descriptive than `true` since users know that versions published after March 24 would require MFA.

**Comment by kevinlinxc** [comment]
@sonalkr132 what do you think of the date solution?

**Comment by sonalkr132** [comment]
Using date would be an improvement. 
The confusing part could be how the info would be interpreted by the user when a much older version page says "requires mfa account since march 24, 2022". 
For example, does it make sense to show mfa requirement on this page https://rubygems.org/gems/rubocop/versions/0.48.0 (when mfa requirement was not a thing) 
Most of info on the version page sidebar is about that specific version. Perhaps we should mfa requirement info only on versions where mfa requirement was met.

**Comment by kevinlinxc** [comment]
@sonalkr132  is my understanding correct:
Right now it's
```
mfa required for latest version -> show true for all versions
mfa not required for latest version -> don't show for all versions
```
and we want
```
mfa required for specific version -> show true for that version
mfa not required for that version -> don't show for that version
```

**Comment by kevinlinxc** [comment]
This was meant to be an easy fix but I thought about it and had some ideas for improvement.

Currently, Rubygems.org displays `Requires MFA Accounts: True` on a gem if the latest version of that gem has MFA enabled, and it displays this on all versions of the gem. 

The previous implementation was to display `Requires MFA Accounts: since x.y.z`, which had a bug, which is why we moved to the current state.

So, there are two pieces of information at play:

1. whether or not you need MFA to release a version of the gem today (we show this now)
2. whether or not a gem version had mfa enabled when released (used to show this)

Rather than replace 1 with 2, I think that both of these pieces of information are good to know, so I think we should keep both.

I'm having a bit of trouble with the wording however, as putting these two things next to each other can lead to contradictions if not done properly e.g. MFA required: true + published with MFA: false

I propose

"New versions require MFA:" for 1 and "Version published with MFA:" for 2. Any thoughts?

I made a PR with these changes here: https://github.com/rubygems/rubygems.org/pull/3079

**Comment by sonalkr132** [comment]
closed by #3079
